cg/meta/report/api.py

```diff
@@ -186,7 +187,7 @@ def get_case_data(
         self,
         case: models.Family,
         analysis: models.Analysis,
-        analysis_metadata: MipAnalysis,
+        analysis_metadata: Union[MipAnalysis, BalsamicAnalysis],
```

Comment:
Since these `pydantic` classes have different structures, I think a parent `Analysis` class won't be able to  encapsulate them in a logical way with the current implementation:

https://github.com/Clinical-Genomics/cg/blob/ccc5d3bf05bc811089d7d3975a175c2083481d3d/cg/models/mip/mip_analysis.py#L10-L17

https://github.com/Clinical-Genomics/cg/blob/ccc5d3bf05bc811089d7d3975a175c2083481d3d/cg/models/balsamic/analysis.py#L13-L22

Reply:
Seems like `BalsamicAnalysis`and `MipAnalysis` are not subclasses of Analysis. I guess you could type hint with BaseModel if you wanted only one return type or keep it as is

Reply:
Imagine you have a bunch of classes like
```
class Square(Shape):
...
```
and one method to fetch the shape
```
def get_shape() -> Union[Square, Circle, Triangle, Oval, Rectangle]
```
According to polymorphism a sub-type is the parent type like "a taxi" is "a car" and we could write 
```
def get_shape() -> Shape
```
as long as it feels like it makes sense. You still have to know what kind of Shape you are dealing with if you want to use any specific functionality that is not available in all subclasses of Shape

Reply:
@patrikgrenfeldt:

Restructured: [d94a4b2](https://github.com/Clinical-Genomics/cg/pull/1446/commits/d94a4b2011295ab6eb0da4b742ad2ff9ab8ee149)

Reply:
👍